src/db/models.rs

```diff
@@ -144,3 +151,26 @@ impl Account {
                                                                 user=self.username).into())
     }
 }
+
+impl Status {
+    pub fn account(&self, db_conn: &Connection) -> QueryResult<Account> {
```

Comment:
This `QueryResult<T>` pattern is nice; we should perhaps look into using `error_chain` and passing these up from the helpers on model structs, instead of a `Result<Option<T>>` or something.

Definitely something to look into, since all the existing helpers basically just `unwrap()` and will panic on database error.

Reply:
I think I want to see something like error_chain for tracebacks, but ultimately I want more useful error messages to be written at the time of the throw.

Maybe a pattern like:

```rust
use err::RustodonError;
do_db_junk(blah).map_err(|e| RustodonError::with("couldn't get account for status", e))?;
```

and then traceback handling code while rocket is in dev mode.

(I wonder if rocket has some error_chain integration. That feels like a really good thing for it to have.)

Reply:
[This post](https://jamesmunns.com/update/2017/07/22/rocket-plus-error-chain.html) on using Rocket with error_chain, and the [comments for that post](https://www.reddit.com/r/rust/comments/6oxk44/using_rocket_error_chain_for_rest_apis_in_rust/) on r/rust are interesting.